### ISSUE TITLE
Achieve same behavior of viewer "page-fit" in render function

### ISSUE THREAD

**Issue by enricodente** [mozilla/pdf.js#12009]
Attach (recommended) or Link to PDF file here:
Rendering PDFs in 16:9 works fine, as they are set to fit on the screen for their width, but other PDFs (like 4:3 or Letter/A4) get cut in height. I would like to get the same behavior of the Viewer "page-fit" mode. Tried multiple variations, but did not succeed.

Configuration:
- Web browser and its version: Any
- Operating system and its version: Any
- PDF.js version: Latest
- Is a browser extension: No

Steps to reproduce the problem:
```
	pdfViewerState.pdf.getPage(pageToReach).then((page) => {
		
		var container = document.getElementById("canvas_container");
		var canvas = document.getElementById("pdf_renderer");
		var ctx = canvas.getContext('2d');
		var viewport = page.getViewport({scale:pdfViewerState.zoom});

		var scale = container.clientWidth / viewport.width;
		viewport = page.getViewport({scale:scale});

		canvas.width = viewport.width;
		canvas.height = viewport.height;

		var renderTask = page.render({
			canvasContext: ctx,
			viewport: viewport
		});

        
```

This is one of the changes I tried, unsuccessful:
```
		const pageWidthScale = container.clientWidth / page.view[2];
		const pageHeightScale = container.clientHeight / page.view[3];
		//console.log('H ' + pageHeightScale);
		//console.log('W ' + pageWidthScale);
		var scale = Math.min(pageWidthScale, pageHeightScale);
		viewport = page.getViewport({scale:scale});
```
Another try:
```
		let scale = Math.min((container.clientWidth / page.view[2]), (container.clientHeight / page.view[3]));
		let viewport = page.getViewport({scale: scale});
		canvas.width = Math.min(container.clientWidth, page.view[2] * scale);
		canvas.height = Math.min(container.clientHeight, page.view[3] * scale);
```
Basically the PDF gets rendered very small in the top left corner.

What is the expected behavior? (add screenshot)
The PDF fits the largest dimension between H and W on the screen

What went wrong? (add screenshot)
I can only achieve width fit.

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by Snuffleupagus** [comment]
> PDF.js version: Latest

Please make sure to always provide an *actual* version number here, since "Latest" can mean any number of things. (Are you e.g. referring to the latest *Stable release*, the latest *Pre-release*, or something else?)

Furthermore, please see https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md (emphasis mine):
> If you are developing a custom solution, first check the examples at https://github.com/mozilla/pdf.js#learning and search existing issues. If this does not help, **please prepare a short well-documented example that demonstrates the problem and make it accessible online** on your website, JS Bin, GitHub, etc. before opening a new issue or contacting us in the Matrix room -- **keep in mind that just code snippets won't help us troubleshoot the problem**.

**Comment by enricodente** [comment]
Hi Snuffleupagus,

thanks for your reply. I am using the latest release, thus v2.5.207.

I have been searching for this already both here and on SO, but with no success. Also, my example is based on the "Interactive examples" page. I also tried to get some hints from the viewer source code: it makes sense, but in the end the code does not seem to work as expected.

**Comment by enricodente** [comment]
If it can be of any help, I solved the issue by relying on window.innerHeight as container.clientHeight did not seem to exist, and that was the cause of the problem.
